### ISSUE TITLE
Add the `Relevant Projects` form section

### ISSUE THREAD

**Issue by amlan-roy** [amlan-roy/resume-craft#15]
**Details**
<img src="https://github.com/amlan-roy/resume-craft/assets/59330872/ebd59daf-2ff3-43f5-a5eb-9c4de729865e" height="300px" />

AC:

- All the input fields should work.
- For the date field, "current"/end date are mutually exclusive (But optional).
  - If current/end date is selected, then start date is required.
- The big delete button on top deletes the whole section.
- The smaller delete button deletes the sub sections.
- The small plus below adds the sub section.

**Are you willing to contribute to this issue?**
Yes

**Additional context**
The Issue https://github.com/amlan-roy/resume-craft/issues/14 should be done before this issue, so that we have a clear idea of how to integrate the dates input.

**Comment by amlan-roy** [comment]
Added the section. Tests remaining.

**Comment by amlan-roy** [comment]
Will create separate issue for tests for this component. Closing this issue.
